functional_annotation/functional_annotation.md

```diff
@@ -0,0 +1,239 @@
+This allows us to take advantage of KEGG pathway information.
+
+`kofamscan` is run by using the `exec_annotation` command:
+
+```
+exec_annotation --help
+```
+
+`kofamscan` needs a few database files in order to run.
+Let's download them.
+
+```
+wget ftp://ftp.genome.jp/pub/db/kofam/ko_list.gz      # download the ko list 
+wget ftp://ftp.genome.jp/pub/db/kofam/profiles.tar.gz # download the hmm profiles
+```
+
+The first file is the KEGG orthlog (KO) list.
+The second file is the HMM profiles for the KOs.
+
+We also have to unzip and untar the relevant files:
+```
+gunzip ko_list.gz
+tar xf profiles.tar.gz
+```
+
+kofamscan runs using a config file. 
+Using nano, build a config file that looks like this:
+
+```
+nano config.yaml
+```
+
+```
+# Path to your KO-HMM database
+# A database can be a .hmm file, a .hal file or a directory in which
+# .hmm files are. Omit the extension if it is .hal or .hmm file
+profile: ./profiles
+
+# Path to the KO list file
+ko_list: ko_list
+
+# Path to an executable file of hmmsearch
+# You do not have to set this if it is in your $PATH
+# hmmsearch: /usr/local/bin/hmmsearch
+
+# Path to an executable file of GNU parallel
+# You do not have to set this if it is in your $PATH
+# parallel: /usr/local/bin/parallel
+
+# Number of hmmsearch processes to be run parallelly
+cpu: 8
+```
+
+**Challenge**
+How many lines contain computer-readable information in this config file?
+
+<details>
+  <summary>Challenge Answer</summary>
+    Three. Every other line is commented out.
+</details>
+
+Let's run the actual annotation command now.
+```
+exec_annotation -f mapper --config config.yaml -o SRR8859675_final_contigs_kofamscan.txt SRR8859675_final_contigs.faa
+```
+
+Look at the results:
+
+```
+less SRR8859675_final_contigs_kofamscan.txt 
+```
+
+## Visualizing the output
+
+```
+mamba create -n keggdecoder python=3.6 pip
+conda activate keggdecoder
+pip install KEGGDecoder==1.2.2
+```
+
+```
+KEGG-decoder -i SRR8859675_final_contigs_kofamscan.txt -o kegg_decoder_out --vizoption static
+```
+
+When this is done running, you'll have a file named `kegg_decoder_out.svg`. 
+You can open this in jupyter notebook or transfer it to your local computer using a program like `scp`.
+We've recreated the visual below so you can look at it!
+
+![](https://i.imgur.com/kB5CuIK.png)
+
+**Challenge**
+How would this visualization change if we had run `kofamscan` on MAGS vs. on the metagenome assembly?
+
+<details>
+  <summary>Challenge Answer</summary>
+    Assuming every contig was binned, we would still see all of the functions represented in the plot, but we would have a row for each MAG. The functions would be spread out across the organisms.
+</details>
+
+## Next steps/future directions
+
+We annotated a thing! Now what?
+The next steps in any workflow will depend on the goal of your study.
+If you have multiple samples, you could do ortholog or pathway level differential abundance analysis with a tool like corncob. 
```

Comment:
I probably wouldn't include this line unless wanting to spend time explaining the difference between what this would be doing (differential abundance of recovered copies) vs getting coverage of these annotations and then using the summed coverage values of each annotation to look at differential abundance of the functional annotations we recovered/assigned. If you follow what i mean

Reply:
removed!